### PR TITLE
change auth behavior to match upstream on unknown/empty user - use null auth

### LINKED ISSUE #1594: kube-rs can't use empty user like Go's kubectl can
### Current and expected behavior

When using kube-rs over `kubectl proxy` kube-rs client doesn't work while kubectl does
```
kubectl config set-cluster testt --server=http://localhost:8001 --insecure-skip-tls-verify=true
kubectl config set-context testt --cluster=testt
kubectl config use-context testt
```

then run `kubectl get pods` (using kube-rs' kubectl example) and it will fail, while official binary works.

```
Error: Failed to infer configuration: failed to infer config: in-cluster: (failed to read an incluster environment variable: environment variable not found), kubeconfig: (failed to find named user: )

Caused by:
    0: failed to infer config: in-cluster: (failed to read an incluster environment variable: environment variable not found), kubeconfig: (failed to find named user: )
    1: failed to find named user: 
```

Related -  https://github.com/metalbear-co/mirrord/issues/2803

### Possible solution

_No response_

### Additional context

_No response_

### Environment

doesn't matter

### Configuration and features

_No response_

### Affected crates

kube-client

### Would you like to work on fixing this bug?

Yes

### PR DESCRIPTION
**Pull request by aviramha**
change auth behavior to match upstream on unknown/empty user - use null auth
Closes #1594
Should help https://github.com/metalbear-co/mirrord/issues/2803

### CODE REVIEW

File: kube-client/src/config/file_loader.rs [hunk 1]
```diff
@@ -83,12 +83,21 @@ impl ConfigLoader {
...
         let user_name = user.unwrap_or(&current_context.user);
+
+        // client-go doesn't fail on empty/missing user, so we don't either
+        // see https://github.com/kube-rs/kube/issues/1594
         let mut user = config
             .auth_infos
             .iter()
             .find(|named_user| &named_user.name == user_name)
             .and_then(|named_user| named_user.auth_info.clone())
-            .ok_or_else(|| KubeconfigError::FindUser(user_name.clone()))?;
```
Comment (clux) [inline kube-client/src/config/file_loader.rs:91]: We should remove this error variant as well in config/mod.

**Review by clux — COMMENTED** [review]
Thank you! Looks sensible to me. Good find.

File: kube-client/src/config/file_loader.rs (hunk 1 above)
Comment (aviramha) [reply kube-client/src/config/file_loader.rs:91]: Good catch

**Comment by aviramha** [comment]
Seems that nightly fails are not related to the changes.

**Comment by clux** [comment]
Yeah, I don't know what's up with nightly here. But it's not worth worrying about atm I think.

**Review by clux — APPROVED** [review]

**Pull request merged**